### PR TITLE
Fix structural error of api.oas3.yaml

### PR DESCRIPTION
**Pull request by Moonded**
Fix api.oas3.yaml
- parameter substring spelling
- components - parameters - Site (required) boolean
- CastVoteStatus - set allOf to object not array.

### CODE REVIEW

File: web/resources/api/api.oas3.yaml [hunk 1]
```diff
@@ -3193,7 +3193,8 @@ components:
       required: [vote]
       properties:
         vote:
-          allOf: ['#/components/schemas/CastVote']
```
Comment (Monkatraz) [inline web/resources/api/api.oas3.yaml:3196]: If you're gonna do formatting changes, make that a separate PR as that's endless bikeshedding. Most of the formatting choices I made were to preserve my sanity as I basically wrote this entire thing myself lol.

**Review by Monkatraz — APPROVED** [review]

File: web/resources/api/api.oas3.yaml (hunk 1 above)
Comment (Monkatraz) [reply web/resources/api/api.oas3.yaml:3196]: Wait I'm a moron, disregard


this is not my brightest moment

**Review by emmiegit — APPROVED** [review]
LGTM

File: web/resources/api/api.oas3.yaml (hunk 1 above)
Comment (Moonded) [reply web/resources/api/api.oas3.yaml:3196]: Gotcha, i only noticed this due to swagger complaining, so i put it all together into one pr :)
